src/snowflake/snowpark/table.py

```diff
@@ -519,9 +531,18 @@ def merge(
             )
         )
         add_api_call(new_df, "Table.update")
-        return _get_merge_result(
-            new_df._internal_collect_with_tag(statement_params=statement_params),
-            inserted=inserted,
-            updated=updated,
-            deleted=deleted,
+        result = new_df._internal_collect_with_tag(
+            statement_params=statement_params,
+            block=block,
+            data_type=f"{str(snowflake.snowpark._AsyncDataType.MERGE.value)}_{int(inserted)}{int(updated)}{int(deleted)}",
```

Comment:
do not pass in the data in this way (alternatively add more enum types, or add one more argument to _internal_collect_with_tag, etc.)